src/app/Fake.Runtime/RuntimeReferenceAssemblies.fs

```diff
@@ -0,0 +1,156 @@
+module Fake.Runtime.RuntimeReferenceAssemblies
+
+open System.IO
+open System.Runtime.InteropServices
+open Fake.Core
+open Fake.IO.FileSystemOperators
+open Fake.DotNet
+open Fake.Runtime
+open Paket
+
+/// here we will pin Fake runner execution framework to .NET 6 as in `PinnedFrameworkRuntimeVersion`
+/// We will also try to resolve the current SDK that runner is executing, if it is the same as pinned
+/// one then we will use runtime assemblies from that SDK version on its installation on disk. Otherwise,
+/// we will default to NetStandard2.0 assemblies. We will download them since they are packaged in a NuGet
+/// package extract them a and reference them.
+type RuntimeReferenceAssemblies() =
```

Comment:
While I think the implementation is fine, I personally find that name a bit confusing.
Historically, I used the following terms: (Disclaimer: I'm not sure if that is technically the most accurate, see https://docs.microsoft.com/en-us/dotnet/standard/assembly/reference-assemblies) 
- Reference Assembly: This term is for assemblies we can only use for compilation
- Runtime Assembly: This indicates assemblies we might need to load dynamically on-demand (and potentially tell the runtime where to find them)

It is obvious that you cannot use Reference Assemblies when running the script (after compilation), because they don't actually contain any code.
It is not so obvious that you should prefer Reference Assemblies when compiling (if they are available in the NuGet package / SDK), IIRC I run into errors as well with some packages when trying to use Runtime Assemblies for compilation. Note: Most NuGet packages don't include reference assemblies, in this case using the runtime assemblies is fine and they will end up in the 'reference assemblies' list (and that is what the NuGet spec says), but all that logic is in Paket.Core. For the SDK we need to do it 'manually' unfortunately/

Given this, I find it confusing to have "RuntimeReferenceAssemblies", my suggestion would be "SdkAssemblyResolver" or something like that?